### PR TITLE
refactor(config) refactor for testability and add tests

### LINKED ISSUE #68: Refactor config code for testability
AC:
* Refactor the config code for testability
* Add integration tests for setting various config values, modeled after the tests in `profiles_integration_test.go`

### PR DESCRIPTION
**Pull request by ctrombley**
Resolves #68 . 

This PR refactors the config code for testability and adds several integration tests.

### CODE REVIEW

File: internal/config/config.go [hunk 1]
```diff
@@ -351,6 +361,12 @@ func (c *Config) visitAllConfigFields(f func(*Value) error) error {
 	// Iterate through the fields in the struct
 	for i := 0; i < cfgType.NumField(); i++ {
 		field := cfgType.Field(i)
+
+		// Skip unexported fields
+		if field.PkgPath != "" {
```
Comment (jthurman42) [inline internal/config/config.go]: interesting trick

File: internal/config/config.go [hunk 2]
```diff
@@ -381,11 +397,12 @@ func unmarshalAllScopes(cfgViper *viper.Viper) (*map[string]Config, error) {
 	return &cfgMap, nil
 }
 
-func readConfig() (*viper.Viper, error) {
+func readConfig(configDir string) (*viper.Viper, error) {
 	cfgViper := viper.New()
 	cfgViper.SetEnvPrefix(DefaultEnvPrefix)
 	cfgViper.SetConfigName(DefaultConfigName)
 	cfgViper.SetConfigType(DefaultConfigType)
+	cfgViper.AddConfigPath(configDir)              // adding home directory as first search path
```
Comment (jthurman42) [inline internal/config/config.go]: Should change the comment here.

Comment (ctrombley) [reply internal/config/config.go]: 👍

**Comment by ctrombley** [comment]
@jthurman42 This issue was to get some testing in.  I'll add another issue for a command line option for setting the config directory.

File: internal/config/config.go (hunk 1 above)
Comment (ctrombley) [reply internal/config/config.go]: https://groups.google.com/forum/#!topic/golang-nuts/i50jwcgHAcs

**Pull request merged**